Small fixes for supported models (#2471)
huggingface/text-generation-inference: docs/source/supported_models.md

- [Idefics](https://huggingface.co/HuggingFaceM4/idefics-9b) (Multimodal)

> If the above list lacks the model you would like to serve, depending on the model's pipeline type, you can try to initialize and serve the model anyways to see how well it performs, but performance isn't guaranteed for non-optimized models. Read more about [Non-core Model Serving](../basic_tutorials/non_core_models).

Reviewer: ```suggestion
If the model you would like to serve is not on the above list, depending on the model's pipeline type, you can try to initialize and serve the model anyway to see how well it performs, but performance isn't guaranteed for non-optimized models. 

```python
 # for causal LMs/text-generation models
 AutoModelForCausalLM.from_pretrained(<model>, device_map="auto")`
 # or, for text-to-text generation models
 AutoModelForSeq2SeqLM.from_pretrained(<model>, device_map="auto")
 ```

 If you wish to serve a supported model that already exists on a local folder, just point to the local folder.

 ```bash
 text-generation-launcher --model-id <PATH-TO-LOCAL-BLOOM>
 ```

Read more about [Non-core Model Serving](../basic_tutorials/non_core_models).
```

I think keeping both would make more sense.

Reviewer: Ah GH messed up the suggestion.

Author: I was removing the python snippet in the PR. Do you want to re-add it?

Reviewer: Yeah - because it ties up nicely with the message that even if the model is not supported you can still use it with TGI, here's a small snippet to test and you can go read it more about it here.

(I don't feel too strongly about it, so feel free to call the shot as you deem fit)

Author: What about the Python snippet?

Reviewer: I think it'd be good to have the Python snippet. but don't feel too strongly. your call! 🤗
